### PR TITLE
Respect control file

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/clients/default_target_proposal.rb [hunk 1]
```diff
@@ -210,7 +211,13 @@ def create
 
       def change_default_target
         self.default_target = ProductFeatures.GetFeature('globals', 'default_target')
-        detect_target
+        if default_target.nil? || default_target.empty?
+          detect_target
+        elsif Targea::SUPPORTED.include?(default_target)
+          log.info " Using target '#{default_target}' from control file."
```
Comment (imobachgs) [inline src/clients/default_target_proposal.rb]: np: whitespace before `Using...`

File: test/default_target_proposal_test.rb [hunk 2]
```diff
@@ -11,4 +11,52 @@
...
+      allow(Yast::Stage).to receive(:initial).and_return(true) # skip state validation
+    end
+
+    context "auto mode" do
+      before do
+        allow(Yast::Mode).to receive(:autoinst).and_return(true)
+      end
+
+      # TODO how to test it?
+      it "keep default target as it is before"
```
Comment (imobachgs) [inline test/default_target_proposal_test.rb]: is -> was

File: test/default_target_proposal_test.rb [hunk 3]
```diff
@@ -11,4 +11,52 @@
...
+      it "sets target level according to control file" do
+        allow(Yast::ProductFeatures).to receive(:GetFeature).with("globals", "default_target")
+          .and_return("multi-user")
+
+        subject.new.call(["MakeProposal"])
+
+        expect(Yast::ServicesManagerTarget.default_target).to eq "multi-user"
+      end
+
+      it "proposes target when control file not specify" do
```
Comment (imobachgs) [inline test/default_target_proposal_test.rb]: "is not specified in control file"

File: test/default_target_proposal_test.rb (hunk 2 above)
Comment (imobachgs) [reply test/default_target_proposal_test.rb]: What about mocking `ServicesManagerTarget.default_target` to some value and checking that it is [assigned](https://github.com/yast/yast-services-manager/blob/ef8b472b777e0f4e9b2d59ae7de46b0b741c0920/src/lib/services-manager/clients/default_target_proposal.rb#L187)?

Comment (jreidinger) [reply test/default_target_proposal_test.rb]: @imobachgs checking that it is assigned is quite hard, as it is accessible only as localized output of proposal output. It is not public reader.

**Pull request merged**